### PR TITLE
fix: drawer failing to cancel move event

### LINKED ISSUE #279: Dialog thinks mouse is pressed when exiting context menu
When I open a context menu and dismiss it by clicking outside of it. The drawer thinks my mouse is pressed and is moving like when I use my mouse to drag to close it.

https://github.com/emilkowalski/vaul/assets/839848/0fb10c73-47d7-416a-84ed-83df5ab55d42

### PR DESCRIPTION
**Pull request by rortan134**
Closes #279

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -808,13 +808,14 @@ const Content = React.forwardRef<HTMLDivElement, ContentProps>(function (
...
   const pointerStartRef = React.useRef<{ x: number; y: number } | null>(null);
+  const lastKnownPointerEventRef = React.useRef<React.PointerEvent<HTMLDivElement> | null>(null);
 
   React.useEffect(() => {
     // Trigger enter animation without using CSS animation
     setVisible(true);
   }, []);
 
-  const isDeltaInDirection = (delta: { x: number; y: number }, direction: DrawerDirection, threshold = 0) => {
+  function isDeltaInDirection(delta: { x: number; y: number }, direction: DrawerDirection, threshold = 0) {
```
Comment (rortan134) [inline src/index.tsx:818]: changed to regular function for the sake of consistency

File: src/index.tsx [hunk 2]
```diff
@@ -1035,12 +1047,9 @@ const Content = React.forwardRef<HTMLDivElement, ContentProps>(function (
...
       }}
-      onPointerUp={(event) => {
-        rest.onPointerUp?.(event);
-        pointerStartRef.current = null;
-        wasBeyondThePointRef.current = false;
-        onRelease(event);
-      }}
+      onPointerUp={handleOnPointerUp}
+      onPointerOut={handleOnPointerUp}
+      onContextMenu={handleOnPointerUp}
```
Comment (joaom00) [inline src/index.tsx:1052]: We need to forward the events to these too

**Comment by joaom00** [comment]
I think the events are not being forwarded yet. I'm talking specifically about this line

```ts
 rest.onPointerUp?.(event);
```

We need to

```ts
rest.onPointerOut?.(event)
rest.onContextMenu?.(event)
```
depending on the prop

**Comment by emilkowalski** [comment]
Added in #440 due to too many merge conflicts, it's my bad that I didn't merge this earlier. Thanks for your contribution though!

**Pull request closed without merging**